SPECS/nbdkit/nbdkit.spec

```diff
@@ -515,157 +401,189 @@ nbdkit-offset-filter       Serve an offset and range.
+export PYTHON=python3
 %configure \
-    PYTHON=%{_bindir}/python3 \
+    --with-extra='%{name}-%{version}-%{release}' \
     --disable-static \
     --disable-golang \
-%if !%{with ruby}
+    --disable-rust \
     --disable-ruby \
-%endif
+    --disable-ssh \
+    --with-libguestfs \
 %if 0%{?have_ocaml}
     --enable-ocaml \
 %else
     --disable-ocaml \
 %endif
-%if 0%{?have_libguestfs}
-    --with-libguestfs \
+%ifarch %{complete_test_arches}
+    --enable-libguestfs-tests \
 %else
-    --without-libguestfs \
+    --disable-libguestfs-tests \
 %endif
     --with-tls-priority=@NBDKIT,SYSTEM
 
+#    --disable-lua \
+#    --disable-perl \
+#    --disable-tcl \
+#    --without-ext2 \
+#    --without-iso \
+#    --without-libvirt
+
 # Verify that it picked the correct version of Python
 # to avoid RHBZ#1404631 happening again silently.
 grep '^PYTHON_VERSION = 3' Makefile
 
-make %{?_smp_mflags}
+%make_build
+
+# Delete the S3 plugin as it is not built
+rm -f %{buildroot}%{_mandir}/man1/nbdkit-S3-plugin*
+rm -f %{buildroot}%{_libdir}/%{name}/plugins/nbdkit-S3-plugin*
 
 
 %install
 %make_install
 
 # Delete libtool crap.
-find $RPM_BUILD_ROOT -name '*.la' -delete
+find %{buildroot} -type f -name "*.la" -delete -print
 
 # If cargo happens to be installed on the machine then the
 # rust plugin is built.  Delete it if this happens.
-rm -f $RPM_BUILD_ROOT%{_mandir}/man3/nbdkit-rust-plugin.3*
+rm -f %{buildroot}%{_mandir}/man3/nbdkit-rust-plugin.3*
 
-# Remove the deprecated ext2 plugin (use ext2 filter instead).
-rm $RPM_BUILD_ROOT%{_libdir}/%{name}/plugins/nbdkit-ext2-plugin.so
-rm $RPM_BUILD_ROOT%{_mandir}/man1/nbdkit-ext2-plugin.1*
+# Delete the S3 plugin as it is not built
+rm -f %{buildroot}%{_mandir}/man1/nbdkit-S3-plugin*
+rm -f %{buildroot}%{_libdir}/%{name}/plugins/nbdkit-S3-plugin*
 
 
 %check
+%ifnarch %{broken_test_arches}
+function skip_test ()
+{
+    for f in "$@"; do
+        rm -f "$f"
+        echo 'exit 77' > "$f"
+        chmod +x "$f"
+    done
+}
+
 # Workaround for broken libvirt (RHBZ#1138604).
 mkdir -p $HOME/.cache/libvirt
 
 # tests/test-captive.sh is racy especially on s390x.  We need to
 # rethink this test upstream.
-truncate -s 0 tests/test-captive.sh
+skip_test tests/test-captive.sh
 
 %ifarch s390x
 # Temporarily kill tests/test-cache-max-size.sh since it fails
 # sometimes on s390x for unclear reasons.
-truncate -s 0 tests/test-cache-max-size.sh
+skip_test tests/test-cache-max-size.sh
 %endif
 
 # Temporarily kill test-nbd-tls.sh and test-nbd-tls-psk.sh
 # https://www.redhat.com/archives/libguestfs/2020-March/msg00191.html
-truncate -s 0 tests/test-nbd-tls.sh tests/test-nbd-tls-psk.sh
+skip_test tests/test-nbd-tls.sh tests/test-nbd-tls-psk.sh
 
-# Make sure we can see the debug messages (RHBZ#1230160).
+# Make sure we can see the debug messages
 export LIBGUESTFS_DEBUG=1
 export LIBGUESTFS_TRACE=1
-
-make %{?_smp_mflags} check
-
+%make_build check || {
+    cat tests/test-suite.log
+    exit 1
+}
```

Comment:
Double-checking: did the test succeed for both architectures after this line was restored?

Reply:
I have cut-pasted screenshot from buddy build.  Is that the subject matter of interest?

![image](https://user-images.githubusercontent.com/15831719/201080444-731d229b-7357-4649-97af-a3908193f267.png)

Reply:
Verifying tests is a bit more tricky at the moment, because a failing `%check` section doesn't fail the whole package build, so one needs to look into the logs and look at the lines with "=== CHECK" in them to verify, if the tests are good or not.

Having said that, I now checked what the tests for the moved packages looked like so far and it `nbdkit` was failing anyway (we need to look into that separately) and the only other spec with a `%check` section was `slirp4netns`, where that section is empty, so there's nothing to be done here when it comes to tests.